29.md

```diff
@@ -162,41 +210,29 @@ If the group is forked and hosted in multiple relays, there will be multiple ver
 
 - *group admins* (`kind:39001`) (optional)
 
-Similar to the group metadata, this event is supposed to be generated by relays that host the group.
-
-Each admin gets a label that is only used for display purposes, and a list of permissions it has are listed afterwards. These permissions can inform client building UI, but ultimately are evaluated by the relay in order to become effective.
+Each admin is listed along with one or more roles. These roles SHOULD have a correspondence with the roles supported by the relay, as advertised by the `kind:39003` event.
 
-The list of capabilities, as defined by this NIP, for now, is the following:
-
-- `add-user`
-- `edit-metadata`
-- `delete-event`
-- `remove-user`
-- `add-permission`
-- `remove-permission`
-- `edit-group-status`
-- `delete-group`
-
-```json
+```jsonc
 {
   "kind": 39001,
   "content": "list of admins for the pizza lovers group",
   "tags": [
     ["d", "<group-id>"],
-    ["p", "<pubkey1-as-hex>", "ceo", "add-user", "edit-metadata", "delete-event", "remove-user"],
-    ["p", "<pubkey2-as-hex>", "secretary", "add-user", "delete-event"]
-  ]
+    ["p", "<pubkey1-as-hex>", "ceo"],
```

Comment:
how can we set admin permissions for each role or admin ?

Reply:
Roles are defined at the relay level in my implementation. For instance there are 3 roles.

owner
admin
moderator

There is a kind 39003 that will describe the roles and what permissions they have. That can be used on the client side to understand what an "admin" can do, etc.

Now right now, my relay only supports 3 roles and every group gets those three roles, but at some point maybe we could have roles that could be chosen by the group creator. I don't want to go down this road now though. 

On the relay side those roles are enforced on what they can do. For example.

An admin can add-user with role no "higher" than itself. An admin can remove a user with role no higher than itself, etc.

A 39003 might look something like this.

```json
{
  "content": "",
  "created_at": 1728676272,
  "id": "121489f55f3e59db7d60751604a1ba76d3896ea4285fa4c6873ee3d59ebe4523",
  "kind": 39003,
  "pubkey": "000009056fd123b6a2368f532ec707f9b368c0d1fed62f68be0baeceec776274",
  "sig": "a3088ef308a4a6ac73d7144692828001d769652d112cf13e8e724d161c55cfe680f0a883735c23bb87966bb76c3fe9c3d481a9cf3d98ffc4a52aa0efcf442ad7",
  "tags": [
    [
      "d",
      "01J9YJ56G6RHSTANHPA2WASDSD"
    ],
    [
      "owner",
      "Owner",
      "The owner role is like god mode for the group",
      "add-user",
      "edit-metadata",
      "delete-event",
      "remove-user",
      "delete-group"
    ],
    [
      "admin",
      "Admin",
      "The admin role can do everything except remove the owner, delete the group and change owner role",
      "add-user",
      "edit-metadata",
      "delete-event",
      "remove-user"
    ],
    [
      "moderator",
      "Moderator",
      "The moderator role can do everything except remove the owner and admin, delete the group, edit metadata and change roles",
      "add-user",
      "delete-event",
      "remove-user"
    ]
  ]
}
```

A client could use this to determine what users could do. Again remember, the roles/permissions are enforced on the relay.

Reply:
> One thing that's not in this PR...

true. that was my point. @fiatjaf need to add permissions somewhere.